HIPs/firefly.md

```diff
@@ -0,0 +1,378 @@
+
+We have the following companies committed to contributing as of this proposal:
+
+- Kaleido: 6 developers to contribute to FireFly, including FireFly core, plugins for Enterprise Ethereum, Fabric and Corda, plugins for secure data exchange, developer CLI tools and administrative console UI (maintaining the current version, developing the new version)
+- Atato: 1 developer to contribute to FireFly, including enhancement to ethconnect and token related features (the same ethconnect is used by both FireFly versions)
+- ConsenSys Health: 1 developer to contribute to FireFly, including developing plugins for Trusted Execution Environments and enhancements critical to the healthcare industry (primarily for the new version)
+
+## How to
+
+_Please note, the tools in this section are still a work in progress at the time of writing._
+
+It is very quick and easy for a developer to get started building an app on top of FireFly using a local development environment. The FireFly CLI will walk a developer through the process. FireFly CLI uses Docker to run many parts of the system, so Docker is a prerequisite for running a local instance of FireFly for development purposes.
+
+[https://github.com/kaleido-io/firefly-cli](https://github.com/kaleido-io/firefly-cli)
+
+### Install FireFly CLI
+
+```
+$ go install github.com/kaleido-io/firefly-cli/ff@latest
+```
+
+After the CLI is installed you can create a new dev stack by running the following command:
+
+```
+$ ff init my-new-stack
+```
+
+This command will ask several questions about which specific plugins you would like to use. It will then create a new stack called “my-new-stack”
+
+To launch the stack, run:
+
+```
+$ ff start my-new-stack
+```
+
+Once all the components in the system have started up, information such as endpoints and port numbers will be printed in the terminal. A link will also be provided to the FireFly Console web interface which can be used to view the environment.
+
+To stop the stack run:
+
+```
+$ ff stop my-new-stack
+```
+
+To completely remove the stack, including all of its data, run:
+
+```
+$ ff remove my-new-stack
+```
+
+For developers looking to make contributions to the FireFly project, please see the GitHub repositories
+
+- [https://github.com/kaleido-io/firefly](https://github.com/kaleido-io/firefly)
+- [https://github.com/kaleido-io/firefly-cli](https://github.com/kaleido-io/firefly-cli)
+- [https://github.com/kaleido-io/firefly-ui](https://github.com/kaleido-io/firefly-ui)
+- [https://github.com/kaleido-io/ethconnect](https://github.com/kaleido-io/ethconnect)
+- [https://github.com/kaleido-io/firefly-dataexchange-https](https://github.com/kaleido-io/firefly-dataexchange-https)
+
+## References
+
+See [citation guide](http://www.chicagomanualofstyle.org/tools_citationguide.html).
+
+## Closure
+
+The goal of FireFly is to speed up adoption of enterprise blockchain. The success of the project can be measured by the number of projects that are built on top of it.
```

Comment:
Another point in this proposal that could help is: How can somebody using FireFly vs somebody not using FireFly still co-work?
i.e. They can still use blockchain layer for the ordering of their transactions. How they interpret the events, read the data out of the FireFly or sent into FireFly is upto another organization that is not using the FireFly abstraction.